tests/virtualization/universal/patch_guests.pm

```diff
@@ -18,27 +18,41 @@ use virt_autotest::common;
 #use virt_autotest::kernel;
 use virt_autotest::utils;
 
+sub reboot_guest {
+    my $guest = shift;
+    record_info("Rebooting $guest");
+    if (get_var("KVM") || get_var("XEN")) {
+        script_run("rm /tmp/guests_ip/$guest");
+        script_run("virsh shutdown $guest");
+        script_retry("virsh domstate $guest|grep 'shut off'", retry => 5);
+        script_run("virsh start $guest");
+        script_retry("test -f /tmp/guests_ip/$guest", retry => 5, delay => 60);
+    } else {
+        script_run("ssh root\@$guest reboot || true", timeout => 10);
+        wait_guest_online($guest);
+    }
+}
+
 sub run {
     my ($self) = @_;
     select_console('root-console');
+    my @guests = keys %virt_autotest::common::guests;
     set_var('MAINT_TEST_REPO', get_var('INCIDENT_REPO'));
     my $host_os_version = get_var('DISTRI') . "s" . lc(get_var('VERSION') =~ s/-//r);
-    foreach my $guest (keys %virt_autotest::common::guests) {
-        if ($guest =~ /$host_os_version/) {
+    foreach my $guest (@guests) {
+        if ($guest eq $host_os_version || $guest eq "${host_os_version}PV" || $guest eq "${host_os_version}HVM") {
             if (check_var('PATCH_WITH_ZYPPER', '1')) {
                 assert_script_run("ssh root\@$guest dmesg --level=emerg,crit,alert,err -tx|sort -o /tmp/${guest}_dmesg_err_before.txt");
                 record_info("Patching $guest");
                 ssh_add_test_repositories "$guest";
                 ssh_fully_patch_system "$guest";
-
-                record_info("Rebooting $guest");
-                script_run("ssh root\@$guest reboot || true", timeout => 10);
-                wait_guest_online($guest);
+                reboot_guest($guest);
```

Comment:
Is not it a workaround for the unfixed bug in sles15sp1 host only? If we make it a normal reboot process for very guest, Will the potential reboot bug like 15sp1 PV guest not be hidden?

Reply:
Agree, seems bug is still hidden...

If we want to find/expose the issue and meanwhile workaround it once it happens, I have a suggestion that we can check its 'virsh dumpxml <vm>' output for <on_reboot> value. Once it does not match `restart`, we record_softfail for the bug, then apply the original method in "ensure_reboot_policy".  HDYT?

Reply:
We will not use "ssh root\@$guest reboot".  We will use "virsh shutdown/start" to reboot.   "15sp1 PV  <on_reboot>" bug won't be triggered regardless of "ensure_reboot_policy".

Reply:
I am wondering why would you need to switch it since the bug is hidden for both ways?

Reply:
> We will not use "ssh root@$guest reboot". We will use "virsh shutdown/start" to reboot. "15sp1 PV <on_reboot>" bug won't be triggered regardless of "ensure_reboot_policy".

This is the point. If you switch to this new way, the bug won't be triggered either. Same effect, then no need to change it , right?  However, in your progress ticket description, you mentioned you wanted to find some other way to disclose the bug. That's why I gave the suggestion.

Reply:
With this change, "ensure_reboot_policy" can be dropped in the future so the bug will be uncovered/disclosed. Whether or not triggering the bug is another topic.  "disclose it" does not mean "trigger it" here.

Reply:
Sorry for misleading you with "trigger it". Actually I meant "disclose it" -- with the new reboot method by "virsh shutdown/start" , the bug is still hidden, rather than uncovered/disclosed. 

In poo#115148, you described like this:
"Currently, The ensure_reboot_policy in waitfor_guests.pm hides the bug. We should uncover it and find a better workaround."  

With current code change, it is not uncovered, or better workaround. 

Actually I do not intend to involve in detailed solution design , as long as the problem is solved as required in the ticket's description, let alone it  is such a small one.  However, technically it is not solved.  I do not want to tout the solution given in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15352#discussion_r943214647. Any other full solution with less code change is welcome. But I am really confused why you do not see it a workable solution for the ticket? With it, bug can ALREADY be disclosed once it happens, and meanwhile the code change is smaller(just add check for output of `virsh dumpxml` for <on_reboot> value and record it with soft_fail once bug reproduces).

Reply:
It's welcome to ask question/suggest on anything:).  I think your point is to make soft_fail to trigger it for 15.1pv, right?  In bug report developer suggests they won't fix it.  Why would I waste code to reproduce it?  I choose to "ignore" it. 

The change here is:
1. Workround on_restart issue like 15.1pv
2. to make the preparation to remove  ensure_reboot_policy in the future 
3. to make the preparation to replace on_restart with on_destroy  in the future like your ctcs virtualization script does.